test/CSharp/CodeCracker.Test/Refactoring/AllowMembersOrderingCodeFixProviderTests.StyleCop.cs

```diff
@@ -0,0 +1,225 @@
+﻿using CodeCracker.Refactoring;
+using System.Threading.Tasks;
+using TestHelper;
+using Xunit;
+namespace CodeCracker.Test.Refactoring
+{
+    public class StyleCopAllowMembersOrderingCodeFixProviderTests : CodeFixTest<AllowMembersOrderingAnalyzer, StyleCopAllowMembersOrderingCodeFixProvider>
+    {
+        [Fact]
+        public async Task StyleCopAllowMembersOrderingShouldAssureMembersOrderByType()
+        {
+            const string source = @"
+            class Foo
+            {
+                public class Class { }
+                public struct Sruct { }
+                void Method() { }
+                public int this[int a] { set { } }
+                public int Property { set { } }
+                public interface Interface { }
+                enum Enum { Enum1 }
+                public event System.Action Event { add { } remove { } }
+                public delegate double Delegate();
+                ~Foo() { }
+                Foo() { }
+                public event System.Action EventField;
+                public int b = 0;
+            }";
+
+            const string expected = @"
+            class Foo
+            {
+                public int b = 0;
+                public event System.Action EventField;
+                Foo() { }
+                ~Foo() { }
+                public delegate double Delegate();
+                public event System.Action Event { add { } remove { } }
+                enum Enum { Enum1 }
+                public interface Interface { }
+                public int Property { set { } }
+                public int this[int a] { set { } }
+                void Method() { }
+                public struct Sruct { }
+                public class Class { }
+            }";
+
+            await VerifyCSharpFixAsync(source, expected);
+        }
+
+        [Fact]
+        public async Task StyleCopAllowMembersOrderingShouldAssureMembersOrderByModifiers()
+        {
+            const string source = @"
+            public class Foo
+            {
+                private int p = 0;
+                protected int q = 0;
+                protected internal int r = 0;
+                internal int s = 0;
+                public int t = 0;
+                static int u = 0;
+                public static int v = 0;
+                const int x = 0;
+                public const int z = 0;
+            }";
+
+            const string expected = @"
+            public class Foo
+            {
+                public const int z = 0;
+                const int x = 0;
+                public static int v = 0;
+                static int u = 0;
+                public int t = 0;
+                internal int s = 0;
+                protected internal int r = 0;
+                protected int q = 0;
+                private int p = 0;
+            }";
```

Comment:
Actually, I didn't find any document where the precedence is specified.
So, I've used the rules descriptions to try to figure it out.

I'm not sure what come first: const or static when the members are fields. I reckon  "ConstantsMustAppearBeforeFields" is quite more imperative than "StaticElementsMustAppearBeforeInstanceElements", also static fields are still fields.